src/Compilers/VisualBasic/Portable/Symbols/Source/SourceFile.vb

```diff
@@ -331,29 +310,67 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         ''' after the import statements have been added to the SourceFile.
         ''' Specifically, constraints are checked for generic type references.
         ''' </summary>
-        Private Shared Sub ValidateImports(memberImports As ImmutableArray(Of NamespaceOrTypeAndImportsClausePosition),
+        Private Shared Sub ValidateImports(compilation As VisualBasicCompilation,
+                                           memberImports As ImmutableArray(Of NamespaceOrTypeAndImportsClausePosition),
                                            memberImportsSyntax As ImmutableArray(Of SyntaxReference),
                                            aliasImportsOpt As IReadOnlyDictionary(Of String, AliasAndImportsClausePosition),
                                            diagnostics As BindingDiagnosticBag)
             ' TODO: Dev10 reports error on specific type parts rather than the import
             ' (reporting error on Object rather than C in C = A(Of Object) for instance).
+            Dim clauseDiagnostics = BindingDiagnosticBag.GetInstance()
```

Comment:
>clauseDiagnostics [](start = 16, length = 17)

nit: Is there a gain from re-using instance this way, rather than simply getting a new instance on every iteration (from within `ValidateImportsClause`)? May be marginally simpler to read.
I was thrown off initially, wondering what this bag was for and why it was discarded...

Reply:
> Is there a gain from re-using instance this way, rather than simply getting a new instance on every iteration 

Avoiding allocating within a loop on every iteration is always a good thing.

---
In reply to: [368147783](https://github.com/dotnet/roslyn/pull/40987#discussion_r368147783) [](ancestors = 368147783)